front/components/assistant/conversation/Conversation.tsx

```diff
@@ -156,18 +157,24 @@ export default function Conversation({
     return <div>No conversation here</div>;
   }
 
+  const messages = conversation.content.map((versionedMessages) => {
+    let m: UserMessageType | AgentMessageType | ContentFragmentType | null =
+      null;
+    for (const x of versionedMessages) {
+      if (!m || x.version > m.version) {
+        m = x;
+      }
+    }
+    if (!m) {
+      throw new Error("Found empty array in conversation content");
+    }
+
+    return m;
+  });
+
   return (
     <div className="pb-24">
-      {conversation.content.map((versionedMessages) => {
-        // Lots of typing because of the reduce which Typescript
```

Comment:
@philipperolet if we can avoid doing this going forward it would be really cool 🙏

Reply:
(to be clear here the undesirable pattern is the `as ...`)